### ISSUE TITLE
Warn user when SQLite3 connection contains no filename

### ISSUE THREAD

**Issue by Dano507** [knex/knex#5216]
# Environment
Knex version: 2.1.0
Database + version: SQLite3 5.0.8 (npm package version)
OS: Arch Linux

# Issue
When the "SQLite3" driver is being used and `connection.filename` is not defined, the 
program will throw a `TypeError` when trying to make queries, which makes it difficult to 
debug a misspelled `filename`.

notice the unrelated `TypeError` when running this:
```js
const knex = require('knex')({
  client: 'sqlite3',
  connection: {
    filenaem: './db.db',  // misspelled "filename"
  },
});


knex.schema.createTable('test', table => {
  table.string('text')
})
.then(e => {
  console.log(e)
  console.log("creation success")
})
.catch(err => {
  console.log("creation error")
  console.log(err);
});

```

# Possible Solutions
The solution I have in mind, is to add a check inside SQLite3's constructor that warns
the developer when `connection.filename` is undefined

`./lib/dialects/sqlite3/index.js`:
```js
class Client_SQLite3 extends Client {
  constructor(config) {
    super(config);
      // other code
      if (typeof config.connection.filename === "undefined") {
        this.logger.warn(
          '"connection.filename" undefined: ' +
          'please specify a database filepath and name ' + 
          '(likely in the require statement), to avoid errors when querying.'
        );
      }
    }
  
  // other code
}
```

thoughts?

**Comment by Dano507** [comment]
Thanks ramos, great work

**Comment by kibertoad** [comment]
Released in 2.2.0
